spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -493,12 +493,30 @@ struct ProductionVisitor
                 pushState(std::move(pstate));
             }
         }
+
+        auto __offsets = builder::member(state().self, "__offsets");
+        auto cur_offset = builder::memberCall(state().cur, "offset", {});
+
+        // Since the offset list is created empty make sure the element for the currrent field is present.
+        auto index = builder()->addTmp("index", builder::integer(field->index()));
+        auto resize = builder()->addWhile(builder::lowerEqual(builder::size(__offsets), index));
```

Comment:
(Looks like a `vector::resize` method would be helpful that preinitializes fields with defaults. But no need to add that right now, this is fine.)

Reply:
Filed ticket for this.